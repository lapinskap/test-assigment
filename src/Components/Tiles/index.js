import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from 'reactstrap';

export default function Tiles({
  config, onTileClick, selected, md, sm,
}) {
  return (
    <div className="grid-menu grid-menu-4col">
      <Row className="no-gutters" data-t1="tiles">
        {config.map(({
          label, icon, id,
        }) => {
          const isSelected = Array.isArray(selected) ? selected.includes(id) : selected === id;
          return (
            <Col md={md} sm={sm} className={`bg-white${isSelected ? ' border-success' : ''}`} key={id}>
              <div
                className={`widget-chart widget-chart-hover ${isSelected ? 'text-success' : 'text-primary'}`}
                role="presentation"
                data-t1={id}
                onClick={onTileClick ? () => onTileClick(id) : null}
              >
                <div className="icon-wrapper rounded-circle">
                  <div className="icon-wrapper-bg bg-primary" />
                  <i className={`${icon}`} />
                </div>
                <div className="widget-numbers-sm">{label}</div>
              </div>
            </Col>
          );
        })}
      </Row>
    </div>
  );
}
Tiles.propTypes = {
  config: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    icon: PropTypes.string,
    label: PropTypes.string.isRequired,
  })).isRequired,
  onTileClick: PropTypes.func,
  md: PropTypes.string,
  sm: PropTypes.string,
  selected: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  ]),
};

Tiles.defaultProps = {
  onTileClick: null,
  selected: null,
  md: '3',
  sm: '6',
};
