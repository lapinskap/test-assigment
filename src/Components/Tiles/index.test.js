import React from 'react';
import EnzymeToJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Tiles from './index';
import '../../tests/setupTests';

describe('<Tiles />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correct snapshot with simple selected value', () => {
    const onTileClickSpy = jest.fn((id) => {
      expect(id).toBe('test_1');
    });
    const config = [
      {
        id: 'test_1',
        label: 'Test 1',
        icon: 'icon_1',
      },
      {
        id: 'test_2',
        label: 'Test 2',
        icon: 'icon_2',
      },
    ];
    const wrapper = shallow(<Tiles config={config} selected="test_1" onTileClick={onTileClickSpy} />);

    wrapper.find('.widget-chart').at(0).prop('onClick')();
    expect(onTileClickSpy).toHaveBeenCalledTimes(1);
    expect(EnzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('renders correct snapshot with array selected value', () => {
    const config = [
      {
        id: 'test_1',
        label: 'Test 1',
        icon: 'icon_1',
      },
      {
        id: 'test_2',
        label: 'Test 2',
        icon: 'icon_2',
      },
    ];
    const wrapper = shallow(<Tiles config={config} selected={['test_1']} />);
    expect(wrapper.find('.widget-chart').at(0).prop('onClick')).toBe(null);
    expect(EnzymeToJson(wrapper)).toMatchSnapshot();
  });
});
