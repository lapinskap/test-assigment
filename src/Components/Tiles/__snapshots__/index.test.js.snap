// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tiles /> renders correct snapshot with array selected value 1`] = `
<div
  className="grid-menu grid-menu-4col"
>
  <Row
    className="no-gutters"
    data-t1="tiles"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      className="bg-white border-success"
      key="test_1"
      md="3"
      sm="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <div
        className="widget-chart widget-chart-hover text-success"
        data-t1="test_1"
        onClick={null}
        role="presentation"
      >
        <div
          className="icon-wrapper rounded-circle"
        >
          <div
            className="icon-wrapper-bg bg-primary"
          />
          <i
            className="icon_1"
          />
        </div>
        <div
          className="widget-numbers-sm"
        >
          Test 1
        </div>
      </div>
    </Col>
    <Col
      className="bg-white"
      key="test_2"
      md="3"
      sm="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <div
        className="widget-chart widget-chart-hover text-primary"
        data-t1="test_2"
        onClick={null}
        role="presentation"
      >
        <div
          className="icon-wrapper rounded-circle"
        >
          <div
            className="icon-wrapper-bg bg-primary"
          />
          <i
            className="icon_2"
          />
        </div>
        <div
          className="widget-numbers-sm"
        >
          Test 2
        </div>
      </div>
    </Col>
  </Row>
</div>
`;

exports[`<Tiles /> renders correct snapshot with simple selected value 1`] = `
<div
  className="grid-menu grid-menu-4col"
>
  <Row
    className="no-gutters"
    data-t1="tiles"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      className="bg-white border-success"
      key="test_1"
      md="3"
      sm="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <div
        className="widget-chart widget-chart-hover text-success"
        data-t1="test_1"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="icon-wrapper rounded-circle"
        >
          <div
            className="icon-wrapper-bg bg-primary"
          />
          <i
            className="icon_1"
          />
        </div>
        <div
          className="widget-numbers-sm"
        >
          Test 1
        </div>
      </div>
    </Col>
    <Col
      className="bg-white"
      key="test_2"
      md="3"
      sm="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <div
        className="widget-chart widget-chart-hover text-primary"
        data-t1="test_2"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="icon-wrapper rounded-circle"
        >
          <div
            className="icon-wrapper-bg bg-primary"
          />
          <i
            className="icon_2"
          />
        </div>
        <div
          className="widget-numbers-sm"
        >
          Test 2
        </div>
      </div>
    </Col>
  </Row>
</div>
`;
